### PR TITLE
ACQ - Updates to derived table "finance_transactions_purchase_orders.sql" for Metadb

### LINKED ISSUE #797: Add expense class to finance_transactions_purchase_orders derived table for Metadb

### PR DESCRIPTION
**Pull request by sbeltaine**
Closes #797
Added transaction_expense_class_code, transaction_expense_class_name, and external_account_number_ext. Please note that finance_transactions_purchase_orders.sql is already included in the runlist.txt file for Metadb.

Documented this update on the 1.7 (Poppy) release notes at https://github.com/folio-org/folio-analytics/wiki/Changes-by-Release

### CODE REVIEW

File: sql_metadb/derived_tables/finance_transaction_purchase_order.sql [hunk 1]
```diff
@@ -30,13 +33,16 @@ SELECT
     jsonb_extract_path_text(po.jsonb, 'order_type') AS po_order_type,
     jsonb_extract_path_text(po.jsonb, 'vendor')::uuid AS po_vendor_id,
     jsonb_extract_path_text(oo.jsonb, 'name') AS po_vendor_name
+
```
Comment (stdombek) [inline sql_metadb/derived_tables/finance_transaction_purchase_order.sql:36]: Empty line can be deleted.

File: sql_metadb/derived_tables/finance_transaction_purchase_order.sql [hunk 2]
```diff
@@ -30,13 +33,16 @@ SELECT
...
 FROM
     folio_finance.transaction AS ft
     LEFT JOIN folio_orders.po_line AS pol ON jsonb_extract_path_text(ft.jsonb, 'encumbrance', 'sourcePoLineId')::uuid = pol.id
     LEFT JOIN folio_orders.purchase_order AS po ON jsonb_extract_path_text(ft.jsonb, 'encumbrance', 'sourcePurchaseOrderId')::uuid = po.id
     LEFT JOIN folio_finance.fund AS ff ON ft.fromfundid = ff.id
     LEFT JOIN folio_finance.budget AS fb ON ft.fromfundid = fb.fundid AND ft.fiscalyearid = fb.fiscalyearid
-    LEFT JOIN folio_organizations. organizations AS oo ON jsonb_extract_path_text(po.jsonb, 'vendor')::uuid = oo.id
+    LEFT JOIN folio_organizations.organizations AS oo ON jsonb_extract_path_text(po.jsonb, 'vendor')::uuid = oo.id
+    LEFT JOIN folio_finance.expense_class__t AS ec ON ec.id = ft.expenseclassid
+
```
Comment (stdombek) [inline sql_metadb/derived_tables/finance_transaction_purchase_order.sql:45]: Empty line can be deleted.

**Review by stdombek — CHANGES_REQUESTED** [review]
All queries:
- [X] PR Title and Description are accurate and thorough
- [X] PR is based on a new branch (not main)
- [X] PR scope is not overly broad
- [X] Query runs without errors
- [X] Query output is correct
- [X] Query logic is clear and well documented
- [X] Query is readable and properly indented
- [X] Table and column names are in all-lowercase
- [X] Quotation marks are used only where necessary
- [X] JSON extraction is in standard form, for example:
      LDP:     t #>> '{f1,f2,f3}'    [for compatibility across LDP 1 & 2]
      Metadb:  jsonb_extract_path_text(t, f1, f2, f3)

Derived tables:
- [X] First line is "--metadb:table" directive, followed by blank line
- [X] User documentation in comment lines, followed by blank line
- [X] File name is listed in `runlist.txt` after dependencies

**Query is very good, only 2 empty lines can be deleted. **

**Comment by stdombek** [comment]
I only see one problem if I'm not wrong. For the query to work there must always be a table for the expense class? Metadb creates the table only if expense classes there? This is the same issue as the finance_invoice_transactions derived table in PR https://github.com/folio-org/folio-analytics/pull/803 We should definitely check this. @nassibnassar ?

**Comment by nassibnassar** [comment]
The `--metadb:require` directive can be used to ensure the required table and column(s) are present: https://metadb.dev/doc/#_metadbrequire

File: sql_metadb/derived_tables/finance_transaction_purchase_order.sql (hunk 1 above)
Comment (sbeltaine) [reply sql_metadb/derived_tables/finance_transaction_purchase_order.sql]: Done.

File: sql_metadb/derived_tables/finance_transaction_purchase_order.sql (hunk 2 above)
Comment (sbeltaine) [reply sql_metadb/derived_tables/finance_transaction_purchase_order.sql]: Done.

**Review by trapido — APPROVED** [review]
The require directive for the expense classes table has been added, so everything looks good!

All queries:
- [x] PR Title and Description are accurate and thorough
- [x] PR is based on a new branch (not main)
- [x] PR scope is not overly broad
- [x] Query runs without errors
- [x] Query output is correct
- [x] Query logic is clear and well documented
- [x] Query is readable and properly indented
- [x] Table and column names are in all-lowercase
- [x] Quotation marks are used only where necessary
- [x] JSON extraction is in standard form, for example:
      LDP:     t #>> '{f1,f2,f3}'    [for compatibility across LDP 1 & 2]
      Metadb:  jsonb_extract_path_text(t, f1, f2, f3)

Derived tables:
- [x] First line is "--metadb:table" directive, followed by blank line
- [x] User documentation in comment lines, followed by blank line
- [x] File name is listed in `runlist.txt` after dependencies

**Review by stdombek — APPROVED** [review]

**Pull request merged**